File: src/main/java/br/com/rfasioli/micrometermetrics/persistence/document/Person.java
package br.com.rfasioli.micrometermetrics.persistence.document;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
@Data
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Person {
  @Id
  @EqualsAndHashCode.Include
  private ObjectId id;

  private String firstName;
  private String sureName;
  private String lastName;
}

File: src/main/kotlin/br/com/rfasioli/micrometermetrics/persistence/document/Person.kt
package br.com.rfasioli.micrometermetrics.persistence.document

import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class Person (
    @Id val id: ObjectId? = null,
    val firstName: String,
    val sureName: String? = null,
    val lastName: String
)
